Anagram.java

```diff
@@ -24,26 +26,79 @@ public static void main(String args[]) {
 			if (!pass) break;
 		}
 		System.out.println(pass ? "test passed" : "test Failed");
-	}  
+	}
 
 	// Returns true if the two given strings are anagrams, false otherwise.
 	public static boolean isAnagram(String str1, String str2) {
-		// Replace the following statement with your code
-		return false;
-	}
+		String main_str = preProcess(str1);
+		String secondary_str = preProcess(str2);
+
+		for (int i = 0; i < main_str.length(); i++){
+			for (int j = 0; j < secondary_str.length(); j++){
+				if (main_str.charAt(i) == secondary_str.charAt(j)){
+
+					secondary_str = secondary_str.replaceFirst(Character.toString(secondary_str.charAt(j)), "");
+
+				}
+			}
+
+		}
+
+		for (int a = 0; a < secondary_str.length(); a++){
+			if (secondary_str.charAt(a) != ' '){
+				return false;
+			}
+		}
+		return true;
+
+		}
 	   
 	// Returns a preprocessed version of the given string: all the letter characters are converted
 	// to lower-case, and all the other characters are deleted, except for spaces, which are left
 	// as is. For example, the string "What? No way!" becomes "whatnoway"
+
 	public static String preProcess(String str) {
-		// Replace the following statement with your code
-		return "";
+		String preProccessedString = "";
+		char test_char;
+
+		for (int i = 0; i < str.length(); i++){
+
+			test_char = str.charAt(i);
+			if (Character.isLetter(test_char)) {
+				test_char = Character.toLowerCase(test_char);
+				preProccessedString += test_char;
+			}else if(test_char == ' '){
+				preProccessedString += test_char;
+			}
+			
+		}
+
+		return preProccessedString;
 	} 
 	   
 	// Returns a random anagram of the given string. The random anagram consists of the same
 	// characters as the given string, re-arranged in a random order. 
 	public static String randomAnagram(String str) {
-		// Replace the following statement with your code
-		return "";
+		//System.out.println("Start of Rand test");
+		//System.out.println("string is:" + str);
+		String play_str = str;
+		String new_anagram = "";
+
+		while (play_str.length() > 0){
+			int rand_num = (int)(Math.random()*(play_str.length()));
+			//System.out.println("Position rand: " + rand_num);
+			if (rand_num < play_str.length())
+				new_anagram = new_anagram + play_str.charAt(rand_num);
+				play_str = play_str.replaceFirst(Character.toString(play_str.charAt(rand_num)), "");
```

Comment:
Both string.replaceFirst() and Character.toString() wasn't taught in recitation or lecture, therefore it is hw guidelines violation. -4